### ISSUE TITLE
Node version not found when running ./script/setup

### ISSUE THREAD

**Issue by karenc** [openstax/cnx-recipes#1033]
# What happened?

The node version was [recently](https://github.com/openstax/cnx-recipes/commit/dd7c00fc839dbf7a9d82a3cdec87cc96df2f8808#diff-bcf7bcf05ac59e51b9adfd7394c810c4) upgraded from `8.9.4` to `10.15.3`.

I got an error from nodenv saying that "the definition is not found".

The available node versions are:

```
$ nodenv install --list | grep 10.15
  0.10.15
  10.15.0
  10.15.1
```

# Reproducibility Details

When I ran `./script/setup`, I get this error:

```
==> Activating Python virtualenv ... OK
==> Installing correct version of node ...
node-build: definition not found: 10.15.3

See all available versions with `nodenv install --list'.

If the version you need is missing, try upgrading node-build:
  
  cd /root/.nodenv/plugins/node-build && git pull && cd -
./script/setup: ERROR: could not run [nodenv install 10.15.3 --skip-existing]
```

This also happens when building the docker container:

```
$ docker build -t openstax/cnx-recipes:test .
Sending build context to Docker daemon  66.48MB
Step 1/17 : FROM openstax/ci-image:latest as build-os-dependencies
 ---> 10100c825d68
Step 2/17 : WORKDIR /code
 ---> Using cache
 ---> 2b3c2d125684
Step 3/17 : RUN apt-get update
 ---> Using cache
 ---> 1290d88d081f
Step 4/17 : RUN apt-get install     libxml2-utils     xsltproc     unzip           # required by fetch-html
 ---> Using cache
 ---> e7965b3583b2
Step 5/17 : COPY .python-version ./
 ---> Using cache
 ---> eba7a231585e
Step 6/17 : RUN pyenv install --skip-existing "$(< .python-version)"
 ---> Using cache
 ---> b47c77c4fd8c
Step 7/17 : COPY .node-version ./
 ---> Using cache
 ---> bee1ac0a091c
Step 8/17 : RUN nodenv install --skip-existing "$(< .node-version)"
 ---> Running in b8c5ec71bf08
node-build: definition not found: 10.15.3

See all available versions with `nodenv install --list'.

If the version you need is missing, try upgrading node-build:

  cd /root/.nodenv/plugins/node-build && git pull && cd -
The command '/bin/sh -c nodenv install --skip-existing "$(< .node-version)"' returned a non-zero code: 1
```

# Environment:

This doesn't happen if I use nvm instead of nodenv.

I'm building and running the docker container from `master` on ubuntu 18.04.

**Comment by philschatz** [comment]
It sounds like this did not run: https://github.com/openstax/cnx-recipes/blob/master/script/bootstrap#L201 . Can you run `LOG_LEVEL=trace ./script/setup` and include any log messages between lines 190 and 201 (ish)?

**Comment by philschatz** [comment]
Can you upgrade nodenv ? In Docker that requires grabbing a newer version of `ci-image` (I'm unfortunately only skilled enough with Docker to shoot myself in the foot)

**Comment by karenc** [comment]
@philschatz `docker pull openstax/ci-image` fixed the node version issue, then I got into this:

```
Step 10/17 : RUN nodenv exec npm install --global yarn
 ---> Running in 2dd8cb42e230
/root/.nodenv/versions/10.15.3/bin/yarn -> /root/.nodenv/versions/10.15.3/lib/node_modules/yarn/bin/yarn.js
/root/.nodenv/versions/10.15.3/bin/yarnpkg -> /root/.nodenv/versions/10.15.3/lib/node_modules/yarn/bin/yarn.js

> yarn@1.15.2 postinstall /root/.nodenv/versions/10.15.3/lib/node_modules/yarn
> /root/.nodenv/versions/10.15.3/lib/node_modules/.hooks/postinstall
sh: 1: /root/.nodenv/versions/10.15.3/lib/node_modules/.hooks/postinstall: Permission denied
npm ERR! code ELIFECYCLE
npm ERR! errno 126
npm ERR! yarn@1.15.2 postinstall: `/root/.nodenv/versions/10.15.3/lib/node_modules/.hooks/postinstall`
npm ERR! Exit status 126
npm ERR!
npm ERR! Failed at the yarn@1.15.2 postinstall script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /root/.npm/_logs/2019-04-24T15_25_44_203Z-debug.log
The command '/bin/sh -c nodenv exec npm install --global yarn' returned a non-zero code: 126
```

I found https://stackoverflow.com/questions/46111738/how-to-install-global-module-in-docker.  Someone explained that when running npm as root (the default user in a docker container) to install a global package, `--unsafe-perm` flag is necessary.  I'll submit a PR.

Thanks for your help!

**Comment by karenc** [comment]
Fixed #1039.
